### DISCUSSION TITLE
Building source to separate images in YAML

### DISCUSSION THREAD

**Discussion by starnutoditopo** [BretFisher/ama#206] [Q&A]
Suppose we are dealing with and architecture similar to the one in assignment 66 (  https://www.udemy.com/course/docker-mastery/learn/lecture/6790714#overview ), that is:

- a user front-end, accessing the db
- a db
- a result application, just for administrator, accessing the db

Developers wrote code to access the db and use the same component both for the user front-end and the administrator application.

In such a situation:
- a docker container is created to build the sources (the image for this container has the SDK installed)
- binaries are then copied to another docker container: this will host the front-end and the result applications (the image for this container will have the RUNTIME installed, but not the SDK); the build machine is then thrown away
- finally a db container is created.

This situation has something I don't like: the same container has both front-end and result application running. I would like the application to run on its own container and the result to run on other containers.

Is there a good way of creating separate containers in YAML files?

**Comment by BretFisher** [comment]
If the same image has two apps in it because they are the same git repo or something, then just change the CMD when you run the 2nd container to launch the different file. 
Also, look into multi-stage images for keeping build tools out of production images.
